lib/ui/painting.dart

```diff
@@ -1418,6 +1418,44 @@ class Canvas extends NativeFieldWrapperClass2 {
   /// entire group can be made transparent using the [saveLayer]'s paint.
   ///
   /// Call [restore] to pop the save stack and apply the paint to the group.
+  ///
+  /// ## Using saveLayer with clips
+  ///
+  /// When a rectanglular clip operation (from [clipRect]) is not axis-aligned
+  /// with the display, or when the clip operation is not rectalinear (e.g.
+  /// because it is a rounded rectangle clip created by [clipRRect] or an
+  /// arbitrarily complicated path clip created by [clipPath]), the edge of the
+  /// clip needs to be anti-aliased.
+  ///
+  /// If two draw calls overlap at the edge of such a clipped region, without
+  /// using [saveLayer], the first drawing will be anti-aliased with the
+  /// background first, and then the second will be anti-aliased with the result
+  /// of blending the first drawing and the background. On the other hand, if
+  /// [saveLayer] is used immediately after establishing the clip, the second
+  /// drawing will cover the first in the layer, and thus the second alone will
+  /// be anti-aliased with the background when the layer is clipped and
+  /// composited (when [restore] is called).
+  ///
+  /// This point is moot if the clip only clips one draw operation.
+  ///
+  /// ## Performance considerations
+  ///
+  /// Generally speaking [saveLayer] is very expensive.
+  ///
+  /// There are a several different hardware architectures for GPUs (graphics
+  /// processing units, the hardware that handles graphics), but most of them
+  /// involve batching commands and reordering them for performance. When layers
+  /// are used, they cause the rendering pipeline to have to switch render
+  /// target (from one layer to another). Render target switches effectively
+  /// flush the GPU's command buffer, which typically means that optimizations
+  /// that one could get with larger batching are lost. Render target switches
+  /// also generate a lot of memory churn because the GPU needs to copy out the
+  /// current frame buffer contents from the part of memory that's optimized for
+  /// writing, and then needs to copy it back in once the previous render target
+  /// (layer) is restored.
+  ///
+  /// For these reasons, creating a new layer ends up being more or less the
+  /// most expensive operation that one can perform with the [Canvas] API.
```

Comment:
I'm not sure this sentence is true.  I'm sure there are more expensive operations that we expose (or will eventually expose), such as convolution image filters.